java-checks/src/main/java/org/sonar/java/checks/spring/TransactionalMethodVisibilityCheck.java

```diff
@@ -38,14 +49,20 @@ public List<Tree.Kind> nodesToVisit() {
   @Override
   public void visitNode(Tree tree) {
     MethodTree method = (MethodTree) tree;
-    if (!method.symbol().isPublic() && hasTransactionalAnnotation(method)) {
-      reportIssue(method.simpleName(), "Make this method \"public\" or remove the \"@Transactional\" annotation");
+    if (!method.symbol().isPublic()) {
+      for(String annSymbol : proxyAnnotations){
+        if(hasAnnotation(method, annSymbol)){
+          reportIssue(
+            method.simpleName(),
+            "Make this method \"public\" or remove the \""+ annShortName.get(annSymbol)+"\" annotation");
+        }
+      }
```

Comment:
Can we write this in a more functional style to limit the nesting?
```suggestion
      proxyAnnotations.stream()
      .filter(annSymbol -> if(hasAnnotation(method, annSymbol))
      .forEach(annSymbol ->
          reportIssue(
            method.simpleName(),
            "Make this method \"public\" or remove the \""+ annShortName.get(annSymbol)+"\" annotation");
        });
      }
```

Reply:
:1: